### ISSUE TITLE
Invalid Semantic Versioning for Snapshots

### ISSUE THREAD

**Issue by robinheidrich** [phpmyadmin/phpmyadmin#19476]
### Describe the bug

When using the [`semver`](https://www.npmjs.com/package/semver) npm package, you could not properly check the version of phpMyAdmin because `6.0+snapshot` is not valid. I would suggest using `6.0.0+snapshot` instead.

The same applies to 5.2+snapshot.

### Example

```js
const semverGt = require('semver/functions/gt');

console.log(semverGt('5.2.1', '6.0+snapshot'));
```

**Output:** TypeError: Invalid Version: 6.0+snapshot

**Comment by williamdes** [comment]
I checked on the demo server and we have `5.2.2-dev+20241226.a92f647526`
What file do you use to extract the version ?

**Comment by robinheidrich** [comment]
> I checked on the [demo server](https://demo.phpmyadmin.net/QA_5_2/index.php?route=/) and we have `5.2.2-dev+20241226.a92f647526` What file do you use to extract the version ?

The one from the phpMyAdmin website

https://files.phpmyadmin.net/snapshots/phpMyAdmin-6.0%2bsnapshot-all-languages.zip

**Comment by williamdes** [comment]
Sure, sorry I was not very clear in my question.
What file do you use in that archive?
There is for example a version in `package.json` or more information in `libraries/Version.php`

**Comment by robinheidrich** [comment]
I use the version from the file name, `phpMyAdmin-6.0+snapshot-all-languages.zip`

Example:

```js
const semverGt = require('semver/functions/gt');

const previousVersion = '5.2.1';
const newVersion = '6.0+snapshot';
const url = `https://files.phpmyadmin.net/snapshots/phpMyAdmin-${version}-all-languages.zip`;

if (semverGt(previousVersion, newVersion)) {
  // Download the ZIP archive
}
```

If there was a download link for e.g. `5.2.2-dev+20241226.a92f647526` that would be helpful.

**Comment by liviuconcioiu** [comment]
Not familiar with `semver`, but here are 4 different ways I would do this:

1. Report it to semver, if `+snapshot` is not recognized
2. Download the archive, read the version inside the archive and rename the archive
3. Build your own API that returns the newest version
4. Attempt to read the version directly from Github API

5.2.2-dev+ date from ['commit']['commiter']['date'] . first 10 chars from sha
```
https://api.github.com/repos/phpmyadmin/phpmyadmin/commits/QA_5_2
```

6.0.0-dev+ date from ['commit']['commiter']['date'] . first 10 chars from sha
```
https://api.github.com/repos/phpmyadmin/phpmyadmin/commits/master
```

**Comment by williamdes** [comment]
Little know this url https://files.phpmyadmin.net/snapshots/phpMyAdmin-6.0%2bsnapshot.json

But we could improve it, the website uses this file to display the snapshot file.

There is too much impact changing the snapshot name, in the snapshot itself there is all the information you may need. But I am okay to add more info to the json file.
what do you think?

**Comment by liviuconcioiu** [comment]
@williamdes I will also be glad if more info will be added to the snapshot file, like a complete version if possible.

I did not know about this file, since isn't in phpMyAdmin source. Probably a note on the website should be added too in the future.

Also, I see some date discrepancies:

https://files.phpmyadmin.net/snapshots/phpMyAdmin-5.2%2bsnapshot.json

```JSON
{
    "date": "2024-12-30T00:03:23+01:00",
    "commit": "a92f64752669ae2a9040a9eab276341c4db8779e"
}
```

But from downloaded archive

```JSON
"version": "5.2.2-dev+20241229.a92f647526"
```

**Comment by williamdes** [comment]
> Also, I see some date discrepancies:

Short commit version vs full commit ?

**Comment by liviuconcioiu** [comment]
> > Also, I see some date discrepancies:
> 
> Short commit version vs full commit ?

No, the dates are different. the website has 2024-12-30, while the date from composer.json is 2024-12-29.

**Comment by williamdes** [comment]
> > > Also, I see some date discrepancies:
> > 
> > 
> > Short commit version vs full commit ?
> 
> No, the dates are different. the json file has 2024-12-30, while the date from composer.json is 2024-12-29.

Okay, nice catch. I add this to my list while reviewing the build script
